grammar/grammar.y

```diff
@@ -309,13 +315,12 @@ PtrTypeStart
 ArrayTypeStart <- LBRACKET Expr (COLON Expr)? RBRACKET
 
 # ContainerDecl specific
-ContainerDeclAuto <- ContainerDeclType LBRACE container_doc_comment? ContainerMembers RBRACE
-
-ContainerDeclType
-    <- KEYWORD_struct
-     / KEYWORD_opaque
-     / KEYWORD_enum (LPAREN Expr RPAREN)?
-     / KEYWORD_union (LPAREN (KEYWORD_enum (LPAREN Expr RPAREN)? / Expr) RPAREN)?
+ContainerDeclAuto
+    <- KEYWORD_struct LBRACE container_doc_comment? ContainerMembers RBRACE
+     / KEYWORD_opaque LBRACE container_doc_comment? ContainerDeclarations RBRACE
+     / KEYWORD_union LPAREN (KEYWORD_enum (LPAREN Expr RPAREN)? / Expr) RPAREN LBRACE container_doc_comment? TaggedUnionMembers RBRACE
+     / KEYWORD_union LBRACE container_doc_comment? ContainerMembers RBRACE
```

Comment:
Bare union fields can't have a tag value assigned and they cannot be comptime just like tagged union fields. We could use a new rule instead of ContainerField here without the trailing `(EQUALS Expr)?`:
```
BareUnionField <- doc_comment? IDENTIFIER COLON (KEYWORD_anytype / TypeExpr) ByteAlign?
 ```

If we do this, we should probably then rename ContainerMembers to StructMembers and ContainerField to StructField.